### DISCUSSION TITLE
Stack Navigator double header reasoning

### DISCUSSION THREAD

**Discussion by pke** [react-navigation/react-navigation#9442] [Q&A] [answered]
Coming back to stack navigators I am always tripping over the fact that nested stack navigators all show their own header.
I know this behaviour can be disabled by screen options.

I wonder why is this the default in the first place? Why would that ever be useful in a navigational context?

**Comment by raajnadar** [comment]
@satya164 I think there should be an option to override the stack bar when we nest multiple stack navs? What do you think?

**Comment by satya164** [comment, marked as answer]
In majority of cases you'd want to show the header from the child navigator and hide the one from parent since that's the one which will show correct title, have proper animations etc. But since React has one way data flow from parent to child, it's not possible to hide the parent header automatically when there's a header in the child navigator. Hence it needs to be done manually.

So it's isn't a case of usefulness, rather feasibility of implementation.

**Reply by pke** [reply]
Can't the parent navigator handle this? By rendering the child screen and handing in delegates that the client navigator uses instead of handling header rendering itself?

**Reply by satya164** [reply]
Not sure what you mean by that. But a parent component doesn't know what's in a child component.

**Reply by pke** [reply]
The parent component can hand in props to its children. In this case a bunch of functions to render the header.  So the child navigator does not render it's header itself but delegates the rendering to the handed in functions.
